src/ServiceControl.Transports.ASBS/ASBSTransportCustomization.cs

```diff
@@ -40,6 +50,44 @@ protected override void AddTransportForPrimaryCore(IServiceCollection services,
             TransportSettings transportSettings)
         {
             services.AddSingleton<IBrokerThroughputQuery, AzureQuery>();
+
+            var connectionSettings = ConnectionStringParser.Parse(transportSettings.ConnectionString);
+            TopicTopology selectedTopology;
+
+            var serviceBusRootNamespace = new SettingsRootNamespace("ServiceControl.Transport.ASBS");
+            if (connectionSettings.TopicName != null)
+            {
+                //Bundle name provided -> use migration topology
+                //Need to explicitly specific events to be published on the single topic
+                selectedTopology = TopicTopology.FromOptions(new MigrationTopologyOptions
+                {
+                    TopicToPublishTo = connectionSettings.TopicName,
+                    TopicToSubscribeOn = connectionSettings.TopicName,
+                    EventsToMigrateMap = [
+                        "ServiceControl.Contracts.CustomCheckFailed",
+                        "ServiceControl.Contracts.CustomCheckSucceeded",
+                        "ServiceControl.Contracts.HeartbeatRestored",
+                        "ServiceControl.Contracts.HeartbeatStopped",
+                        "ServiceControl.Contracts.FailedMessagesArchived",
+                        "ServiceControl.Contracts.FailedMessagesUnArchived",
+                        "ServiceControl.Contracts.MessageFailed",
+                        "ServiceControl.Contracts.MessageFailureResolvedByRetry",
+                        "ServiceControl.Contracts.MessageFailureResolvedManually"
```

Comment:
It seems those two are above the character counts of 50 chars and would have been shortened to the MD5 hash. The rest is fine.

Furthermore, there might be cases when the ServiceControl endpoint name is longer than 50 chars we would have shortened it automatically into an MD5 hash. We might need to keep this consistent at least for the migration topology and probably discuss what we do for the new topology

Reply:
> It seems those two are above the character counts of 50 chars and would have been shortened to the MD5 hash

Why would that matter on the publisher side? We are not creating rules here.

Reply:
> Furthermore, there might be cases when the ServiceControl endpoint name is longer than 50 chars we would have shortened it automatically into an MD5 hash

Again, doesn't it matter only for the subscribers? The publisher just ensures the _bundle_ topic is present.

Reply:
It is a good point. As of now ServiceControl did never subscribe and therefore it doesn't matter as long as we don't subscribe to any of our integration events